Update the Block List documentation to show how to add an angularJS controller (#2781)
umbraco/UmbracoDocs: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md

If you'd like to display properties of `settings`, you can access these by `block.settingsData.myPropertyAlias`.

### Adding custom implementation to your View

To achieve this you need to add a custom angularJS controller to your custom view, using the ng-controller attribute:

> ```

Reviewer: ```suggestion
```html
```

<div ng-controller="customBlockController" ng-click="block.edit()"> 
    <h2 ng-bind="block.data.headline"></h2>
    <p ng-bind="block.data.description"></p>
</div>
```

Create a folder inside the App_Plugins folder called 'CustomBlockView' or something more meaningful for your implementation.

> Inside this folder create two files, one called package.manifest that contains the following:

Reviewer: ```suggestion
Inside this folder create two files,. The first file should be called package.manifest that contain the following:
```

> ```

Reviewer: ```suggestion
```json
```

{
  "javascript": [
    "~/App_Plugins/CustomBlockView/customBlock.controller.js"    
  ]  
}
```
Umbraco will parse all package.manifest files and load any resources they reference into the backoffice during startup.

> and secondly add a javascript file called

Reviewer: ```suggestion
The second file should be a javascript file called
```

**customBlock.controller.js**

(to match the file referenced in your package.manifest file)

> this file then should register your 'customBlockController' mentioned in your view ng-controller attribute with Umbraco's angular module's controllers

Reviewer: ```suggestion
This file then should register your 'customBlockController' mentioned in your view ng-controller attribute with Umbraco's angular module's controllers
```

> ```

Reviewer: ```suggestion
```javascript
```

angular.module("umbraco").controller("customBlockController", function ($scope) {
// you can do your custom functionality here!
}
```
#### Example: Displaying an image from a Media Picker

> Your block may enable you to 'pick' an image for use as the background for a particular block or to display as part of the block layout, but if you try to display this image directly in the view from the property `block.data.image` you'll see the unique id and not the image.

Reviewer: Sentence was a tat too long - tried splitting it up. Hope it still makes sense 🤞

```suggestion
Your block may enable you to 'pick' an image for use as the background for a particular block or to display as part of the block layout. If you try to display this image directly in the view from the property `block.data.image` you'll see the unique id and not the image.
```

We'll need to use the Id in our custom angularJS controller to get the ImageUrl to display in our Backoffice Block Editor View.

With the setup of files above, we would amend our customBlock.controller.js file, injecting the mediaResource to retrieve the image from the id:

> ```

Reviewer: ```suggestion
```javascript
```

angular.module("umbraco").controller("customBlockController", function ($scope, mediaResource) {

    //your property is called image so the following will contain the udi:
        var imageUdi = $scope.block.data.image;
    //the mediaResource has a getById method:
        mediaResource.getById(imageUdi)
            .then(function (media) {
                console.log(media);
                //set a property on the 'scope' called imageUrl for the returned media object's mediaLink
                $scope.imageUrl = media.MediaLink;
    });    
});
```
Update the View to use the 'imageUrl' property to display the image:

> ```

Reviewer: ```suggestion
```html
```

<div ng-controller="customBlockController" ng-click="block.edit()">
    <h2 ng-bind="block.data.headline"></h2>
    <img src="{{imageUrl}}" />
    <p ng-bind="block.data.description"></p>
</div>
```
If you need to use a specific crop, you can inject the `imageUrlGeneratorResource` resource, which has a `getCropUrl(mediaPath, width, height, imageCropMode, animationProcessMode)` method:

> ```

Reviewer: ```suggestion
```javascript
```
